File: backend/app/schemas/sport.py
from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime


class SportBase(BaseModel):
    name: str
    code: str
    description: Optional[str] = None
    available_assessments: List[str] = []
    is_active: bool = True


class SportCreate(SportBase):
    pass


class SportUpdate(BaseModel):
    name: Optional[str] = None
    code: Optional[str] = None
    description: Optional[str] = None
    available_assessments: Optional[List[str]] = None
    is_active: Optional[bool] = None


class SportResponse(SportBase):
    id: int
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class SportListResponse(BaseModel):
    id: int
    name: str
    code: str
    available_assessments: List[str]
    is_active: bool

    class Config:
        from_attributes = True

File: frontend/src/types/sport.ts
import { AssessmentType } from './assessment';

export interface Sport {
  id: number;
  name: string;
  code: string;
  description: string | null;
  available_assessments: AssessmentType[];
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface SportListItem {
  id: number;
  name: string;
  code: string;
  available_assessments: AssessmentType[];
  is_active: boolean;
}

export interface SportCreate {
  name: string;
  code: string;
  description?: string | null;
  available_assessments: AssessmentType[];
  is_active?: boolean;
}

export interface SportUpdate extends Partial<SportCreate> {}
